v21.2/deploy-app-netlify.md

```diff
@@ -0,0 +1,227 @@
+{{site.data.alerts.end}}
+
+## Step 4. Initialize Prisma Client
+
+1. Set the local `DATABASE_URL` environment variable to a valid connection string to the `leaderboard` database on your cluster:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ export DATABASE_URL="postgresql://<username>:<password>@<host>:<port>/leaderboard?sslmode=verify-full&options=--cluster%3D<routing-id>"
+    ~~~
+
+    Where:
+    - `<username>` is the [SQL user](authorization.html#sql-users) connecting to the cluster.
+    - `<password>` is the password for the SQL user connecting to the cluster.
+    - `<host>` is the host on which the CockroachDB node is running.
+    - `<port>` is the port at which the CockroachDB node is listening.
+    - `<routing-id>` is cluster's routing ID (e.g., `funky-skunk-123`). The routing ID identifies your tenant cluster on a [multi-tenant host](../cockroachcloud/architecture.html#architecture).
+
+    {{site.data.alerts.callout_info}}
+    This connection string is identical to the connection string you obtained earlier from the {{ site.data.products.db }} Console, with the exception of the database parameter (`leaderboard`).
+    {{site.data.alerts.end}}
+
+     By default, Prisma uses the local `DATABASE_URL` environment variable as the connection string to the database.
+
+1. Initialize Prisma Client:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ npx prisma generate
+    ~~~
+
+    Your application can now communicate with your CockroachDB cluster using Prisma Client.
+
+## Step 5. Deploy the application
+
+You can deploy web applications directly from GitHub to Netlify. In this tutorial, we use the [Netlify CLI](https://docs.netlify.com/cli) to deploy the app.
+
+1. Sign into your Netlify account:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ netlify login
+    ~~~
+
+1. Run the app server locally to preview your site:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ netlify dev
+    ~~~
+
+    If the local deployment succeeds, you should see the following message in your terminal:
+
+    ~~~
+    ┌─────────────────────────────────────────────────┐
+    │                                                 │
+    │   ◈ Server now ready on http://localhost:8888   │
+    │                                                 │
+    └─────────────────────────────────────────────────┘
+    ~~~
+
+    For a preview of the site, visit [http://localhost:8888](http://localhost:8888).
+
+1. From your repo directory, deploy your app with the Netlify CLI:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ netlify init
+    ~~~
+
+    Select the default options for each of the prompts. You will be required to authorize Netlify with GitHub.
```

Comment:
Wouldn't most people want to create a new site instead of using an existing one? Or does the site get created earlier?

Reply:
And what about the build command? Mine defaulted to no build command, which doesn't seem right.

Reply:
I got stuck here. Do we need to instruct users to fork the repo b/c Netlify wants access to an actual GitHub repo?

Reply:
```
netlify init
? What would you like to do? +  Create & configure a new site
? Team: kernfeld-cockroach's team
Choose a unique site name (e.g. super-cool-site-by-kernfeld-cockroach.netlify.app) or leave it blank for a random name. You can update the site name later.
? Site name (optional): undefined

Site Created

Admin URL: https://app.netlify.com/sites/super-cool-site-by-kernfeld-cockroach
URL:       https://super-cool-site-by-kernfeld-cockroach.netlify.app
Site ID:   85829de0-cef3-41ff-80d5-67124f3868ec

Linked to super-cool-site-by-kernfeld-cockroach in /Users/paulkernfeld/projects/cockroachdb-typescript/.netlify/state.json

? Netlify CLI needs access to your GitHub account to configure Webhooks and Deploy Keys. What would you like to do? Authorize wit
h GitHub through app.netlify.com
? Your build command (hugo build/yarn run build/etc): # no build command
? Directory to deploy (blank for current dir): build
? Netlify functions folder:
Adding deploy key to repository...
 ›   Error: Failed adding GitHub deploy key with error: Not Found. Does the repository cockroachdb-typescript exist and do cockroachdb has the correct permissions to set up deploy keys?
 ```

Reply:
Forking seemed to work. I did `gh repo fork --remote`.